Fix broken py3 versions/test_views.py tests

we added skipif for:
[TestDownloadSource.test_owner_should_be_allowed](https://github.com/mozilla/addons-server/tree/master/src/olympia/versions/tests/test_views.py#L510) and
[TestDownloadSource.test_group_binarysource_should_be_allowed](https://github.com/mozilla/addons-server/tree/master/src/olympia/versions/tests/test_views.py#L539) but both should, in theory, pass on python3.  We need to investigate and fix.

Both are failing because the filename encoding is coming out of the header differently than expected by doing just `force_bytes(filename)`

filename is `b'crosswarpex-\xed\x99\x95\xec\x9e\xa5-2.1.072-src.zip'` and the header is `b'attachment; filename="crosswarpex-\xc3\xad\xc2\x99\xc2\x95\xc3\xac\xc2\x9e\xc2\xa5-2.1.072-src.zip"'` (both forced to bytes for clarity over the actual data)

The error is returned as:
```
___________________________________ TestDownloadSource.test_owner_should_be_allowed ____________________________________
src/olympia/versions/tests/test_views.py:520: in test_owner_should_be_allowed
    assert filename in response['Content-Disposition']
E   AssertionError: assert 'crosswarpex-확장-2.1.072-src.zip' in 'attachment; filename="crosswarpex-í\x99\x95ì\x9e¥-2.1.072-src.zip"'
```

from debugging where the header is set it's clear that its `b'attachment; filename="crosswarpex-\xed\x99\x95\xec\x9e\xa5-2.1.072-src.zip"'` when we set the header; and `b'attachment; filename="crosswarpex-\xc3\xad\xc2\x99\xc2\x95\xc3\xac\xc2\x9e\xc2\xa5-2.1.072-src.zip"'` by the time we read it in the test.  I.e. `\xed\` becomes `\xc3\xad` and `\x99\` becomes `\xc2\x99` (I guess because `í` ends up as `\xed\`, the unicode code point `U+00ED`, but the literal UTF-8 encoding is `\xc3\xad`)

Similar to the middleware test fails, there's clearly something _wrong_, but it's not completely clear where the fault is - or if it would even happen outside of our artificial test suite.

--- a/src/olympia/versions/tests/test_views.py
+++ b/src/olympia/versions/tests/test_views.py
@@ -3,6 +3,7 @@ import os
 import six
 
 from django.conf import settings
+from django.utils.encoding import smart_text
 from django.core.files import temp
 from django.core.files.base import File as DjangoFile
 from django.test.utils import override_settings
@@ -23,6 +24,19 @@ from olympia.amo.urlresolvers import reverse
 from olympia.amo.utils import urlencode, urlparams
 from olympia.files.models import File
 from olympia.users.models import UserProfile
+
+
+def decode_http_header_value(value):
+    """
+    Reverse the encoding that django applies to bytestrings in
+    HttpResponse._convert_to_charset(). Needed to test header values that we
+    explicitly pass as bytes such as filenames for content-disposition or
+    xsendfile headers.
+    """
+    if six.PY3:
+        return value.encode('latin-1').decode('utf-8')
+    else:
+        return value.decode('utf-8')
 
 
 class TestViews(TestCase):
@@ -446,23 +460,19 @@ class TestDownloadSource(TestCase):
         )
         self.url = reverse('downloads.source', args=(self.version.pk, ))
 
-    @pytest.mark.skipif(
-        six.PY3,
-        reason='Currently broken in Python 3, needs to be fixed')
     def test_owner_should_be_allowed(self):
         self.client.login(email=self.user.email)
         response = self.client.get(self.url)
         assert response.status_code == 200
         assert response[settings.XSENDFILE_HEADER]
         assert 'Content-Disposition' in response
-        filename = self.filename
-        if not isinstance(filename, six.text_type):
-            filename = filename.decode('utf8')
-        assert filename in response['Content-Disposition'].decode('utf8')
-        path = self.version.source.path
-        if not isinstance(path, six.text_type):
-            path = path.decode('utf8')
-        assert response[settings.XSENDFILE_HEADER].decode('utf8') == path
+        filename = smart_text(self.filename)
+        content_disposition = response['Content-Disposition']
+        assert filename in decode_http_header_value(content_disposition)
+        expected_path = smart_text(self.version.source.path)
+        xsendfile_header = decode_http_header_value(
+            response[settings.XSENDFILE_HEADER])
+        assert xsendfile_header == expected_path
 
     def test_anonymous_should_not_be_allowed(self):
         response = self.client.get(self.url)
@@ -475,9 +485,6 @@ class TestDownloadSource(TestCase):
         response = self.client.get(self.url)
         assert response.status_code == 404
 
-    @pytest.mark.skipif(
-        six.PY3,
-        reason='Currently broken in Python 3, needs to be fixed')
     def test_group_binarysource_should_be_allowed(self):
         GroupUser.objects.create(user=self.user, group=self.group)
         self.client.login(email=self.user.email)
@@ -485,14 +492,13 @@ class TestDownloadSource(TestCase):
         assert response.status_code == 200
         assert response[settings.XSENDFILE_HEADER]
         assert 'Content-Disposition' in response
-        filename = self.filename
-        if not isinstance(filename, six.text_type):
-            filename = filename.decode('utf8')
-        assert filename in response['Content-Disposition'].decode('utf8')
-        path = self.version.source.path
-        if not isinstance(path, six.text_type):
-            path = path.decode('utf8')
-        assert response[settings.XSENDFILE_HEADER].decode('utf8') == path
+        filename = smart_text(self.filename)
+        content_disposition = response['Content-Disposition']
+        assert filename in decode_http_header_value(content_disposition)
+        expected_path = smart_text(self.version.source.path)
+        xsendfile_header = decode_http_header_value(
+            response[settings.XSENDFILE_HEADER])
+        assert xsendfile_header == expected_path
 
     def test_no_source_should_go_in_404(self):
         self.version.source = None

